Dockerfile-linux.template

```
@@ -45,9 +45,11 @@ RUN set -eux; \
 	\
 # download/install js-yaml
 	mkdir -p /opt/js-yaml/; \
-	wget -O /opt/js-yaml/js-yaml.js "https://github.com/nodeca/js-yaml/raw/${JSYAML_VERSION}/dist/js-yaml.js"; \
-	wget -O /opt/js-yaml/package.json "https://github.com/nodeca/js-yaml/raw/${JSYAML_VERSION}/package.json"; \
-	ln -s /opt/js-yaml/js-yaml.js /js-yaml.js; \
+	wget -O /opt/js-yaml/js-yaml.tgz https://registry.npmjs.org/js-yaml/-/js-yaml-${JSYAML_VERSION}.tgz; \
+	echo "662e32319bdd378e91f67578e56a34954b0a2e33aca11d70ab9f4826af24b941  /opt/js-yaml/js-yaml.tgz" | sha256sum -c -; \
+	tar -xz --strip-components=1 -f /opt/js-yaml/js-yaml.tgz -C /opt/js-yaml; \
```

Perhaps only extract the dist/js-yaml.js file?

---

We also need `package.json` right?  I don't remember whether the note on `strip-components` about "Pathnames with fewer elements will be silently skipped." means it skips processing them entirely or just skips stripping them, but I'm hoping it's the latter so something like this would be sufficient?

```suggestion
	tar -xz --strip-components=1 -f /opt/js-yaml/js-yaml.tgz -C /opt/js-yaml dist/js-yaml.js package.json; \
```

---

(Then we could drop the non-tarball bits of the following `rm`)

---

Yeah, I missed [that line](https://github.com/docker-library/mongo/pull/700/files#diff-b3c0fcf605d397bd3fce8ea61588d2c9da8b24a9fe6886b2e640d97914f1d01fL49)

---

Regarding the strip components, it looks like the stripping happens when writing but not when interpreting the list of files to extract from the command line. So you need

```
tar -xz --strip-components=1 -f /opt/js-yaml/js-yaml.tgz -C /opt/js-yaml package/dist/js-yaml.js package/package.json
```

and you end up with

```
/opt/js-yaml/package.json
/opt/js-yaml/dist/js-yaml.js
```

which is the [new line 52](https://github.com/docker-library/mongo/pull/700/files#diff-b3c0fcf605d397bd3fce8ea61588d2c9da8b24a9fe6886b2e640d97914f1d01fR52) requires.

---

Ah, but it would probably muck up `package.json`'s path if we did `--strip-components=2` (which if it was otherwise more aggressive would give us the same folder structure as we have currently without the added `dist`) :disappointed:

I think the other options for fixing _that_ are `--transform` (which is murky), `mv`/`rmdir`, or multiple `tar` invocations, none of which is ideal either. :thinking: